### PR TITLE
PYTHON-2234 Only rely on standard library in daemon.py

### PR DESCRIPTION
**Pull request by prashantmital**
Error message after this change:

```
/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py:137: RuntimeWarning: Failed to start mongocryptd: is it on your $PATH?
Original exception: [Errno 2] No such file or directory: 'mongocryptd': 'mongocryptd'
  _silence_resource_warning(_spawn(sys.argv[1:]))
/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py:137: RuntimeWarning: Failed to start mongocryptd: is it on your $PATH?
Original exception: [Errno 2] No such file or directory: 'mongocryptd': 'mongocryptd'
  _silence_resource_warning(_spawn(sys.argv[1:]))
```

### CODE REVIEW

File: pymongo/daemon.py [hunk 1]
```diff
@@ -64,8 +65,10 @@ def _spawn_daemon(args):
                     stdin=devnull, stderr=devnull, stdout=devnull)
                 _silence_resource_warning(popen)
         except FileNotFoundError as exc:
-            raise PyMongoError(
-                f'Failed to start {args[0]}: is it on your $PATH?\nOriginal exception: {exc}')
+            warnings.warn(f'Failed to start {args[0]}: is it on your $PATH?\n'
+                          f'Original exception: {exc}', RuntimeWarning,
+                          stacklevel=2)
+            return subprocess.CompletedProcess(args, 0)
```
Comment (prashantmital) [inline pymongo/daemon.py:71]: Without this, the error is very convoluted:
```
 /Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py:137: RuntimeWarning: Failed to start mongocryptd: is it on your $PATH?
Original exception: [Errno 2] No such file or directory: 'mongocryptd': 'mongocryptd'
  _silence_resource_warning(_spawn(sys.argv[1:]))
Traceback (most recent call last):
  File "/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py", line 137, in <module>
    _silence_resource_warning(_spawn(sys.argv[1:]))
  File "/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py", line 51, in _silence_resource_warning
    popen.returncode = 0
AttributeError: 'NoneType' object has no attribute 'returncode'
/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py:137: RuntimeWarning: Failed to start mongocryptd: is it on your $PATH?
Original exception: [Errno 2] No such file or directory: 'mongocryptd': 'mongocryptd'
  _silence_resource_warning(_spawn(sys.argv[1:]))
Traceback (most recent call last):
  File "/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py", line 137, in <module>
    _silence_resource_warning(_spawn(sys.argv[1:]))
  File "/Users/pmital/Developer/mongo-python-driver/pymongo/daemon.py", line 51, in _silence_resource_warning
    popen.returncode = 0
AttributeError: 'NoneType' object has no attribute 'returncode'
```

Comment (ShaneHarvey) [inline pymongo/daemon.py:71]: What's the point of `return subprocess.CompletedProcess(args, 0)`? The success case doesn't return anything.

**Review by ShaneHarvey — COMMENTED** [review]
Can you post some output of the new behavior?

File: pymongo/daemon.py (hunk 1 above)
Comment (prashantmital) [reply pymongo/daemon.py:71]: See my comments above... The success case returns the output of `subprocess.Popen`.

Comment (prashantmital) [reply pymongo/daemon.py:71]: Another option is to check that `_spawn` returned non-`None` before trying to set the `returncode` in `suppress_resource_warning`

Comment (ShaneHarvey) [reply pymongo/daemon.py:71]: Look again, `_spawn_daemon` (Windows) doesn't return anything on success, but `_spawn` (unix) does. It's odd to return `CompletedProcess` in the first case.

Comment (prashantmital) [reply pymongo/daemon.py]: Ah got it. I changed it so that `_silence_resource_warning` only sets returncode when it gets a non-None `Popen`.

**Review by ShaneHarvey — APPROVED** [review]
LGTM, let's wait for evergreen to pass.

**Pull request merged**